api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -827,21 +815,185 @@ private String generateRandomEightCharacterQualifier() {
+    final String matrixName = "mt_" + qualifier;
+
+    return ImmutableList.of(
+        "# Note: Hail demos must be run from a \"Hail Genomics Analysis\" cloud analysis environment",
+        generateExtractionDirCode(qualifier, dataSetExportRequest),
+        "# Initialize Hail\n"
+            + "\n"
+            + "import hail as hl\n"
+            + "import os\n"
+            + "from hail.plot import show\n"
+            + "\n"
+            + "hl.init(default_reference='GRCh38')\n"
+            + "hl.plot.output_notebook()",
+        "# Create Hail Matrix table\n"
+            + "# This can take a few hours for larger (several hundred participant) cohorts\n"
+            + "workspace_bucket = os.environ['WORKSPACE_BUCKET']\n"
+            + "vcf_dir = os.environ['"
+            + generateVcfDirEnvName(qualifier)
+            + "']\n"
+            + "hail_matrix_table_gcs = f'{workspace_bucket}/dataset_"
+            + qualifier
+            + ".mt'\n"
+            + "hl.import_vcf(f'{vcf_dir}/*.vcf.gz', force_bgz=True).write(hail_matrix_table_gcs)\n"
+            + matrixName
+            + " = hl.read_matrix_table(hail_matrix_table_gcs)",
+        "# Select variants\n" + matrixName + ".rows().select().show(5)",
+        "# Select sample names\n" + matrixName + ".s.show(5)");
+  }
+
+  private List<String> generatePlinkCode(
+      String qualifier, DataSetExportRequest dataSetExportRequest) {
+    final String localVcfDir = "dataset_" + qualifier + "_vcfs";
+    final String mergedVcfFilename = "dataset_" + qualifier + "_merged.vcf.gz";
+    final String mergedVcfFilepath = localVcfDir + "/" + mergedVcfFilename;
+    final String plinkBinaryPrefix = "dataset_" + qualifier + "_plink";
+
+    List<String> plinkCode =
+        ImmutableList.of(
+            "# Create a single merged VCF file\n"
+                + "# This can take a few hours for larger (several hundred participant) cohorts\n"
+                + "\n"
+                + "!bcftools concat -a"
+                + localVcfDir
+                + "/*.vcf.gz -o "
+                + mergedVcfFilepath,
+            "!plink --vcf " + mergedVcfFilepath + " --make-bed --out " + plinkBinaryPrefix,
+            "# Plink binary input files. Optionally - delete "
+                + localVcfDir
+                + "/ if you plan to only use Plink\n"
+                + "# and no longer need the VCF files\n"
+                + "!ls dataset_43789957_plink.*");
+
+    return Stream.concat(
+            generateDownloadVcfCode(qualifier, dataSetExportRequest).stream(), plinkCode.stream())
+        .collect(Collectors.toList());
+  }
+
+  private List<String> generateDownloadVcfCode(
+      String qualifier, DataSetExportRequest dataSetExportRequest) {
+    final String localVcfDir = "dataset_" + qualifier + "_vcfs";
+
+    return ImmutableList.of(
+        generateExtractionDirCode(qualifier, dataSetExportRequest),
+        "%%bash\n"
```

Comment:
Possible future optimization (not for this PR): check there's sufficient disk space for this operation